cmd/device.go

```diff
@@ -141,45 +151,54 @@ func configUpdateHandler(wg *sync.WaitGroup, ping *client.AgentPing, wsm *WebSoc
 			if wsm.IsInitialized && (newDeviceConfig.Enabled == false || newDeviceConfig.Host == "") {
 				wsm.CloseConnection()
 			}
-			handleDeviceUpdate(ping, newDeviceConfig)
+			handleDeviceUpdate(beat, wsm.Credentials, newDeviceConfig)
 		}
 	}
 }
 
-// runDeviceConfigPinger sends pings to service and manages config updates
-func runDeviceConfigPinger(wg *sync.WaitGroup, ping *client.AgentPing, wsm *WebSocketManager, apiOrigin string) {
+// sendDeviceHeartbeats sends device heartbeat messages to the backend api, and receives config updates
+func sendDeviceHeartbeats(wg *sync.WaitGroup, beat *client.DeviceHeartbeat, wsm *WebSocketManager, apiOrigin string) {
 	defer wg.Done()
-	log.Info("Starting agent ping server")
+	log.Info("Sending device heartbeats")
 
 	for {
-		// If the device isn't connected to an audio server, there is no socket connection to the api server so use a regular ping endpoint.
-		if !wsm.IsInitialized || currentDeviceConfig.Host == "" {
-			newDeviceConfig, err := sendPing(*ping, apiOrigin)
+		if currentDeviceConfig.Enabled == true && currentDeviceConfig.Host != "" {
+			// device is connected to an audio server
+
+			// Initialize a socket connection (do nothing if already connected)
+			wsm.InitConnection(wg, beat.MAC) // no need for error check since failure defaults to http heartbeat/config exchange
```

Comment:
Do we always assume that the socket connection would work to the audio device? Previously, I used `!wsm.IsInitialized` check to default back to http pings in case where socket connection isn't working due to any changes (accidental change to load balancers, server updates, etc).

Reply:
I understand `wsm` is managing the heartbeat -> control plane messages, which should always work (after we roll latest to prod). ping fallback from websockets to icmp is handled separately within `MeasurePingStats`.